src/Gui/MainWindow.cpp

```diff
@@ -1938,24 +1933,11 @@ void MainWindow::startSplasher()
 
 void MainWindow::stopSplasher()
 {
-    const auto isWayland = qGuiApp->platformName() == QLatin1String("wayland");
-    if (isWayland) {
-        if (d->splashscreen) {
-            d->splashscreen->finish(this);
-            d->splashscreen->deleteLater();
-            d->splashscreen = nullptr;
-        }
-        return;
+    if (d->splashscreen) {
+        d->splashscreen->finish(this);
+        d->splashscreen->deleteLater();
+        d->splashscreen = nullptr;
```

Comment:
Please make it true revert! There was `delete d->splashscreen;` before. It segfaults with messages enabled.

Reply:
Ok. Quite frustrating. Something is really wrong with this code.